src/megaclient.cpp

```diff
@@ -10178,39 +10193,62 @@ bool MegaClient::startxfer(direction_t d, File* f, bool skipdupes)
             if (it != cachedtransfers[d].end())
             {
                 LOG_debug << "Resumable transfer detected";
-                Transfer *transfer = it->second;
+                t = it->second;
+                if ((d == GET && !t->pos) || ((time(NULL) - t->lastaccesstime) >= 86400))
+                {
+                    LOG_warn << "Discarding temporary URL (" << t->pos << ", " << t->lastaccesstime << ")";
+                    t->cachedtempurl.clear();
+                }
+
                 FileAccess* fa = fsaccess->newfileaccess();
-                if ((fa->fopen(&transfer->localfilename)
-                        && ((d == GET) ||
-                            (d == PUT && (time(NULL) - transfer->lastaccesstime) < 86400
-                                      && !transfer->genfingerprint(fa))))
-                     || (d == GET && !transfer->progresscompleted))
+                if (!fa->fopen(&t->localfilename))
                 {
-                    if (!transfer->pos)
+                    if (d == PUT)
                     {
-                        // to prevent HTTP 403 errors
-                        transfer->cachedtempurl.clear();
+                        LOG_warn << "Local file not found";
+                        // the transfer will be retried to ensure that the file
+                        // is not just just temporarily blocked
+                    }
+                    else
+                    {
+                        if (t->progresscompleted)
+                        {
+                            LOG_warn << "Temporary file not found";
+                            t->chunkmacs.clear();
+                            t->progresscompleted = 0;
+                            t->pos = 0;
+                        }
                     }
-                    LOG_debug << "Resuming transfer";
-                    t = transfer;
                 }
                 else
                 {
-                    if (d == GET)
+                    if (d == PUT)
                     {
-                        LOG_warn << "Temporary file not found";
+                        if (t->genfingerprint(fa))
```

Comment:
`genfingerprint` checks the fingerprint but also updates the Transfer object with the new one. Since we are going to resume the transfer anyway (to preserve its priority) we need that, after the check, the Transfer object contains the updated fingerprint.